src/schema.jl

```diff
@@ -166,12 +166,25 @@ julia> concrete_term(term(:a), (a = [1, 2, 3], b = [0.0, 0.5, 1.0]))
 a(continuous)
 ```
 """
-concrete_term(t::Term, d, hints::Dict{Symbol}) =
-    concrete_term(t, d, get(hints, t.sym, nothing))
-concrete_term(t::Term, dt::ColumnTable, hint) =
-    concrete_term(t, getproperty(dt, t.sym), hint)
-concrete_term(t::Term, dt::ColumnTable, hints::Dict{Symbol}) =
-    concrete_term(t, getproperty(dt, t.sym), get(hints, t.sym, nothing))
+concrete_term(t::Term, d, hints::Dict{Symbol}) = concrete_term(t, d, get(hints, t.sym, nothing))
+
+function concrete_term(t::Term, dt::ColumnTable, hint)
+    msg :: String = checkcol( dt, t.sym )
+    if msg != ""
+        throw(ArgumentError(msg))
+    end
+    return concrete_term(t, getproperty(dt, t.sym), hint)
+end
```

Comment:
style: (also, we don't use type asserts like this elsewhere in the code base and the comparison below is enough to imply that it should be a string, so might just delete it)
```suggestion
function concrete_term(t::Term, dt::ColumnTable, hint)
    msg::String = checkcol( dt, t.sym )
    if msg != ""
        throw(ArgumentError(msg))
    end
    return concrete_term(t, getproperty(dt, t.sym), hint)
end
```